src/main/java/org/spongepowered/api/resource/ResourceManager.java

```diff
@@ -45,7 +49,7 @@
      * @param path The path to the resource
      * @return The resource
      */
-    Optional<Resource> getResource(ResourcePath path);
+    Resource getResource(ResourcePath path) throws IOException;
```

Comment:
Maybe rename this to `openResource` and add a method `hasResource` to check whether a resource exists with a given path.